buku

```diff
@@ -3351,18 +3360,23 @@ def import_html(html_soup, add_parent_folder_as_tag, newtag):
             desc = comment_tag.find(text=True, recursive=False)
 
         # add parent folder as tag
-        if add_parent_folder_as_tag:
-            # could be its folder or not
-            possible_folder = tag.find_previous('h3')
-            # get list of tags within that folder
-            tag_list = tag.parent.parent.find_parent('dl')
-
-            if ((possible_folder) and possible_folder.parent in list(tag_list.parents)):
-                # then it's the folder of this bookmark
-                if tag.has_attr('tags'):
-                    tag['tags'] += (DELIM + possible_folder.text)
-                else:
-                    tag['tags'] = possible_folder.text
+        if add_parent_folder_as_tag is not None:
```

Comment:
i still have opinion that old method still kept as is

so 
```python
if add_parent_folder_as_tag and not use_nested_structure:
    # old method
    ...
elif add_parent_folder_as_tag and use_nested_structure:
    # new method
    ...
```
this keep change recognizable and it is easier if old method is replaced/changed in future

Reply:
Sure, but i have no idea why would the old method change in future. It has clear goal of finding the parent folder,
which btw the new method does safer and more readable.

Reply:
i think it is just (old + what was working and used by user) vs (new + safer + readable)

i will let @jarun decide at the end

but if it is decided that old method will be replace completely, do note it at changelog

Reply:
Okay, i'll wait. BTW if one bookmark is listed twice under different folders, 
both folders structure considered and respective tags added, Kind of neat.